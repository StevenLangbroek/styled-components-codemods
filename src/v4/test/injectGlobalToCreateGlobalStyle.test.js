"use strict";

const pluginTester = require("babel-plugin-tester");
const { injectGlobalToCreateGlobalStyle } = require("..");
const prettier = require("prettier");

const tests = [
  {
    title: "TaggedTemplateExpression case",
    code: `
    import { injectGlobalStyle } from 'styled-components';
    injectGlobalStyle\`\`
    `
  }
];

pluginTester({
  pluginName: "styled-components-v3-to-v4-global-style",
  plugin: injectGlobalToCreateGlobalStyle,
  snapshot: true,
  formatResult: output =>
    prettier.format(output, {
      semi: true,
      singleQuote: true,
      parser: "babylon"
    }),
  tests: [...tests]
});
