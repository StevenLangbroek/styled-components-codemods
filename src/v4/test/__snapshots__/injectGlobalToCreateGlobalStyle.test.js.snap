// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-components-v3-to-v4-global-style TaggedTemplateExpression case: TaggedTemplateExpression case 1`] = `
"
import { injectGlobalStyle } from 'styled-components';
injectGlobalStyle\`\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { createGlobalStyle } from 'styled-components';
const GlobalStyle = createGlobalStyle\`\`;
throw new Error(
  'An injectGlobal usage was converted to createGlobalStyles via codemod but needs to be hooked up. See styled-components.com/docs/api#createglobalstyle for instructions.'
);

"
`;
